From d140335f1ef0321508345484c8fad6d0d330471c Mon Sep 17 00:00:00 2001
From: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date: Tue, 6 Feb 2018 12:14:46 +0100
Subject: [PATCH] OS-6586 remove bogus casts from PCI_CAP_PUTx and
 PCI_XCAP_PUTx definitions Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
 Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

---
 usr/src/uts/common/sys/pci_cap.h | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/usr/src/uts/common/sys/pci_cap.h b/usr/src/uts/common/sys/pci_cap.h
index 730e10d77b..9804913241 100644
--- a/usr/src/uts/common/sys/pci_cap.h
+++ b/usr/src/uts/common/sys/pci_cap.h
@@ -82,12 +82,12 @@ typedef enum {
 #define	PCI_CAP_GET32(h, i, b, o) ((uint32_t) \
 	pci_cap_get(h, PCI_CAP_CFGSZ_32, i, b, o))
 
-#define	PCI_CAP_PUT8(h, i, b, o, d) ((uint8_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_8, i, b, o, d))
-#define	PCI_CAP_PUT16(h, i, b, o, d) ((uint16_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_16, i, b, o, d))
-#define	PCI_CAP_PUT32(h, i, b, o, d) ((uint32_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_32, i, b, o, d))
+#define	PCI_CAP_PUT8(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_8, i, b, o, d)
+#define	PCI_CAP_PUT16(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_16, i, b, o, d)
+#define	PCI_CAP_PUT32(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_32, i, b, o, d)
 
 #define	PCI_XCAP_GET8(h, i, b, o) ((uint8_t) \
 	pci_cap_get(h, PCI_CAP_CFGSZ_8, PCI_CAP_XCFG_SPC(i), b, o))
@@ -96,12 +96,12 @@ typedef enum {
 #define	PCI_XCAP_GET32(h, i, b, o) ((uint32_t) \
 	pci_cap_get(h, PCI_CAP_CFGSZ_32, PCI_CAP_XCFG_SPC(i), b, o))
 
-#define	PCI_XCAP_PUT8(h, i, b, o, d) ((uint8_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_8, PCI_CAP_XCFG_SPC(i), b, o, d))
-#define	PCI_XCAP_PUT16(h, i, b, o, d) ((uint16_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_16, PCI_CAP_XCFG_SPC(i), b, o, d))
-#define	PCI_XCAP_PUT32(h, i, b, o, d) ((uint32_t) \
-	pci_cap_put(h, PCI_CAP_CFGSZ_32, PCI_CAP_XCFG_SPC(i), b, o, d))
+#define	PCI_XCAP_PUT8(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_8, PCI_CAP_XCFG_SPC(i), b, o, d)
+#define	PCI_XCAP_PUT16(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_16, PCI_CAP_XCFG_SPC(i), b, o, d)
+#define	PCI_XCAP_PUT32(h, i, b, o, d) \
+	pci_cap_put(h, PCI_CAP_CFGSZ_32, PCI_CAP_XCFG_SPC(i), b, o, d)
 
 
 extern int pci_cap_probe(ddi_acc_handle_t h, uint16_t index,
-- 
2.21.0

